modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/parse/NamedColorConverterFactory.java

```diff
@@ -0,0 +1,436 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ * 
+ *    (C) 2017, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.mbstyle.parse;
+
+import java.awt.Color;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.geotools.factory.Hints;
+import org.geotools.util.Converter;
+import org.geotools.util.ConverterFactory;
+
+/**
+ * Converter factory ssupporting web or css color names, requires the use of
+ * {@link Hints#COLOR_NAMES} hint with "WEB", "CSS", "X11", or "HTML".
+ * <p>
+ * There are a few subtle differences between WEB, CSS and X11 colors. this and CSS Colors (for the
```

Comment:
Not a big deal but could use a slight comment edit for clarity. I understand what you're saying, but I think the formatting broke a sentence here: "this and CSS colors (for the example ...)"

At the same time, might as well remove the duplicate "the the" on the next line also and fix "ssupporting" at the top.